geonode/base/auth.py

```diff
@@ -170,7 +171,10 @@ def get_token_from_auth_header(auth_header, create_if_not_exists=False):
         if user and user.is_active:
             return get_auth_token(user) if not create_if_not_exists else get_or_create_token(user)
     elif re.search('Bearer', auth_header, re.IGNORECASE):
-        return re.compile(re.escape('Bearer '), re.IGNORECASE).sub('', auth_header)
+        token = re.compile(re.escape('Bearer '), re.IGNORECASE).sub('', auth_header)
+        access_token = AccessToken.objects.get(token=token)
```

Comment:
as above, better to use the `first()` and check it the token is not `None` or use first the `exists()`